### PR TITLE
Update note to provide Pull Provider feature

### PR DESCRIPTION
**Pull request by bangelic**
In the Upgrading Content Hosts section,
the note was updated to provide a
reference about the pull-based provider
feature so customers can try it.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.4/Katello 4.6
* [x] Foreman 3.3/Katello 4.5
* [x] Foreman 3.2/Katello 4.4
* [ ] Foreman 3.1/Katello 4.3
* For Foreman 3.0 or older, please create a separate PR.
* We do not accept PRs for Foreman 2.3 or older.

### CODE REVIEW

File: guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc [hunk 1]
```diff
@@ -6,6 +6,7 @@ The {project-client-name} repository provides `katello-agent` and `katello-host-
 NOTE: The Katello agent is deprecated and will be removed in a future {Project} version.
 Migrate your workloads to use the remote execution feature to update clients remotely.
 For more information, see {ManagingHostsDocURL}Migrating_From_Katello_Agent_to_Remote_Execution_managing-hosts[Migrating from Katello Agent to Remote Execution] in _{ManagingHostsDocTitle}_.
+With the release of {Project} {ProjectVersion}, we are providing support for a pull-based provider, an agent-based tool as a replacement for the Katello agent.
```
Comment (Lennonka) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc:9]:
```suggestion
With the release of {Project} {ProjectVersion}, we provide support for a pull-based client as a replacement for the Katello agent.
```

**Review by Lennonka — CHANGES_REQUESTED** [review]
I'd suggest to simplify the wording.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (Lennonka) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc:9]: It is and this is supposed to be an additional mention in Upgrading.

Comment (adamlazik1) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_content_hosts.adoc:9]: I would drop {ProjectVersion} altogether. It simplifies and shortens the sentence a lot.
```suggestion
{Project} now supports a pull-based provider as a replacement for the Katello agent.
```

**Review by Lennonka — APPROVED** [review]
Merging

**Comment by Lennonka** [comment]
Cherry-picked:
- 0f95ded43..4aecd9c54  3.4 -> 3.4
- 0df310494..5e0099559  3.3 -> 3.3

I don't think this should be cherry-picked to 3.2, because we were adding the pull provider in 3.3 and newer.

**Comment by ekohl** [comment]
> I don't think this should be cherry-picked to 3.2, because we were adding the pull provider in 3.3 and newer.

And upstream 3.2 is EOL and there are no downstream releases. I'd say that nowadays if we pick, we should only pick to 3.2 if it's also picked to 3.1 to keep things consistent (so users don't go from 3.1 with version B to 3.2 with version A to 3.3 with version B.
